sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/scripting/LongJavaScriptCheck.java

```diff
@@ -28,39 +28,46 @@ public class LongJavaScriptCheck extends AbstractPageCheck {
 
   private static final int DEFAULT_MAX_LINES = 5;
 
+  private final StringBuilder text = new StringBuilder();
+
+  private TagNode scriptNode;
+
+  private int startingNodeLinePosition;
+
   @RuleProperty(
     key = "maxLines",
     description = "Max Lines (Number)",
     defaultValue = "" + DEFAULT_MAX_LINES)
   public int maxLines = DEFAULT_MAX_LINES;
 
-  private int linesOfCode;
-  private TagNode scriptNode;
+  @Override
+  public void startElement(TagNode node) {
+    if ("script".equalsIgnoreCase(node.getNodeName())) {
+      scriptNode = node;
+      startingNodeLinePosition = node.getStartLinePosition();
```

Comment:
Given that we already store the `node` as a field, we don't have to store the start line too, as we can simply use `scriptNode.getStartLinePosition()` in the `endElement` method when needed.